src/main/java/fr/sncf/osrd/train/ActivateRoute.java

```diff
@@ -9,44 +9,47 @@
 
 public class ActivateRoute {
     /** This function try to reserve forwarding routes */
-    public static void reserveRoutes(
-            Simulation sim,
-            SignalNavigatePhase.State navigatePhaseState
-    ) throws SimulationError {
-        // TODO have a smarter way to reserve routes
-        if (navigatePhaseState.getRouteIndex() + 1 >= navigatePhaseState.phase.routePath.size())
-            return;
-        var nextRoute = navigatePhaseState.phase.routePath.get(navigatePhaseState.getRouteIndex() + 1);
-        var nextRouteState = sim.infraState.getRouteState(nextRoute.index);
-        // Try to reserve the route if possible
-        if (nextRouteState.status == RouteStatus.FREE)
-            nextRouteState.reserve(sim);
+    public static void reserveRoutes(Simulation sim, SignalNavigatePhase.State navigatePhaseState, Train train)
+            throws SimulationError {
+        for (int advance = 1; advance <= 1; advance++) {
```

Comment:
I'm not entirely sure of what is happening here. If it is the intended behavior, a comment would help.